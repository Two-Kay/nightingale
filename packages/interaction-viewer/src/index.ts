import InteractionViewer from "./interaction-viewer";

if (window.customElements) {
  customElements.define(InteractionViewer.is, InteractionViewer);
}
