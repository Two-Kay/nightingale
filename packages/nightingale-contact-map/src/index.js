import NightingaleContactMap from "./nightingale-contact-map";

const loadComponent = () => {
  customElements.define("nightingale-contact-map", NightingaleContactMap);
};

// Conditional loading of polyfill
if (window.customElements) {
  loadComponent();
} else {
  document.addEventListener("WebComponentsReady", () => {
    loadComponent();
  });
}
