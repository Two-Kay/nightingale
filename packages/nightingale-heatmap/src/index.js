import NightingaleHeatmap from "./nightingale-heatmap";

const loadComponent = () => {
  customElements.define("nightingale-heatmap", NightingaleHeatmap);
};

// Conditional loading of polyfill
if (window.customElements) {
  loadComponent();
} else {
  document.addEventListener("WebComponentsReady", () => {
    loadComponent();
  });
}
