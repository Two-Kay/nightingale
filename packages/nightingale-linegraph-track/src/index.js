import NightingaleLineGraphTrack from "./nightingale-linegraph-track";

const loadComponent = () => {
  customElements.define(
    "nightingale-linegraph-track",
    NightingaleLineGraphTrack
  );
};

// Conditional loading of polyfill
if (window.customElements) {
  loadComponent();
} else {
  document.addEventListener("WebComponentsReady", () => {
    loadComponent();
  });
}
