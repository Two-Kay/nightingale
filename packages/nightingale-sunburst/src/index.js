import NightingaleSunburst from "./nightingale-sunburst";
if (window.customElements) {
  customElements.define("nightingale-sunburst", NightingaleSunburst);
}
export default NightingaleSunburst;
