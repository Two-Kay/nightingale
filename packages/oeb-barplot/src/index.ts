import OebBarplot from "./oeb-barplot";
export default OebBarplot;
