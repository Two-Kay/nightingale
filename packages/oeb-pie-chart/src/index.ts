import OebPieChart from "./oeb-pie-chart";
export default OebPieChart;
