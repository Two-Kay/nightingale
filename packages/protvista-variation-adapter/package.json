{
  "name": "protvista-variation-adapter",
  "version": "3.5.7",
  "description": "Data adapter for variation data",
  "files": [
    "dist",
    "types",
    "src"
  ],
  "scripts": {
    "build:es": "tsc"
  },
  "main": "dist/protvista-variation-adapter.js",
  "module": "dist/es/protvista-variation-adapter.js",
  "types": "dist/es/protvista-variation-adapter.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/ebi-webcomponents/nightingale.git"
  },
  "bugs": {
    "url": "https://github.com/ebi-webcomponents/nightingale/issues"
  },
  "homepage": "https://ebi-webcomponents.github.io/nightingale/",
  "keywords": [
    "nightingale",
    "customelements",
    "webcomponents",
    "variation"
  ],
  "author": "Xavier Watkins &lt;xwatkins@ebi.ac.uk&gt;",
  "license": "MIT",
  "gitHead": "726779864966d11f966f337f2dd84aa4f58d944a",
  "dependencies": {
    "protvista-feature-adapter": "^3.5.7",
    "protvista-variation": "^3.5.7",
    "uuid": "^8.3.1"
  },
  "devDependencies": {
    "@types/uuid": "^8.3.0"
  }
}
