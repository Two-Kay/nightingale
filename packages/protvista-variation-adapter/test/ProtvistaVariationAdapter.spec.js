import { transformData } from "../src/protvista-variation-adapter";
import variants from "./variants.json";

describe("ProtvistaVariationAdapter", () => {
  it("should transform the data adequately", () => {
    const data = transformData(variants);
    expect(data).toMatchSnapshot();
  });
});
