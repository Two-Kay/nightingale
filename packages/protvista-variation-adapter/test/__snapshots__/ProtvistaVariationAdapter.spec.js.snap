// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtvistaVariationAdapter should transform the data adequately 1`] = `
Object {
  "sequence": "MLPGLALLLLAAWTARALEVPTDGNAGLLAEPQIAMFCGRLNMHMNVQNGKWDSDPSGTKTCIDTKEGILQYCQEVYPELQITNVVEANQPVTIQNWCKRGRKQCKTHPHFVIPYRCLVGEFVSDALLVPDKCKFLHQERMDVCETHLHWHTVAKETCSEKSTNLHDYGMLLPCGIDKFRGVEFVCCPLAEESDNVDSADAEEDDSDVWWGGADTDYADGSEDKVVEVAEEEEVAEVEEEEADDDEDDEDGDEVEEEAEEPYEEATERTTSIATTTTTTTESVEEVVREVCSEQAETGPCRAMISRWYFDVTEGKCAPFFYGGCGGNRNNFDTEEYCMAVCGSAMSQSLLKTTQEPLARDPVKLPTTAASTPDAVDKYLETPGDENEHAHFQKAKERLEAKHRERMSQVMREWEEAERQAKNLPKADKKAVIQHFQEKVESLEQEAANERQQLVETHMARVEAMLNDRRRLALENYITALQAVPPRPRHVFNMLKKYVRAEQKDRQHTLKHFEHVRMVDPKKAAQIRSQVMTHLRVIYERMNQSLSLLYNVPAVAEEIQDEVDELLQKEQNYSDDVLANMISEPRISYGNDALMPSLTETKTTVELLPVNGEFSLDDLQPWHSFGADSVPANTENEVEPVDARPAADRGLTTRPGSGLTNIKTEEISEVKMDAEFRHDSGYEVHHQKLVFFAEDVGSNKGAIIGLMVGGVVIATVIVITLVMLKKKQYTSIHHGVVEVDAAVTPEERHLSKMQQNGYENPTYKFFEQMQN",
  "variants": Array [
    Object {
      "accession": "NC_000021.9:g.26112051C>A",
      "alternativeSequence": "N",
      "association": Array [
        Object {
          "dbReferences": Array [
            Object {
              "id": "104300",
              "name": "MIM",
              "url": "https://www.omim.org/entry/104300",
            },
          ],
          "disease": true,
          "evidences": Array [
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/20301340",
                "id": "20301340",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/20301340",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/20301414",
                "id": "20301414",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/20301414",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/20831773",
                "id": "20831773",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/20831773",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/21500874",
                "id": "21500874",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/21500874",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://europepmc.org/abstract/MED/24829003",
                "id": "24829003",
                "name": "pubmed",
                "url": "https://www.ncbi.nlm.nih.gov/pubmed/24829003",
              },
            },
            Object {
              "code": "ECO:0000313",
              "source": Object {
                "alternativeUrl": "https://www.ensembl.org/homo_sapiens/Variation/Explore?v=RCV000813633",
                "id": "RCV000813633",
                "name": "ClinVar",
                "url": "https://www.ncbi.nlm.nih.gov/clinvar/RCV000813633",
              },
            },
          ],
          "name": "Alzheimer disease (AD)",
        },
      ],
      "begin": "51",
      "clinicalSignificances": Array [
        Object {
          "sources": Array [
            "ClinVar",
          ],
          "type": "Variant of uncertain significance",
        },
      ],
      "codon": "AAG/AAT",
      "consequenceType": "missense",
      "cytogeneticBand": "21q21.3",
      "data": Object {
        "alternativeSequence": "N",
        "association": Array [
          Object {
            "dbReferences": Array [
              Object {
                "id": "104300",
                "name": "MIM",
                "url": "https://www.omim.org/entry/104300",
              },
            ],
            "disease": true,
            "evidences": Array [
              Object {
                "code": "ECO:0000313",
                "source": Object {
                  "alternativeUrl": "https://europepmc.org/abstract/MED/20301340",
                  "id": "20301340",
                  "name": "pubmed",
                  "url": "https://www.ncbi.nlm.nih.gov/pubmed/20301340",
                },
              },
              Object {
                "code": "ECO:0000313",
                "source": Object {
                  "alternativeUrl": "https://europepmc.org/abstract/MED/20301414",
                  "id": "20301414",
                  "name": "pubmed",
                  "url": "https://www.ncbi.nlm.nih.gov/pubmed/20301414",
                },
              },
              Object {
                "code": "ECO:0000313",
                "source": Object {
                  "alternativeUrl": "https://europepmc.org/abstract/MED/20831773",
                  "id": "20831773",
                  "name": "pubmed",
                  "url": "https://www.ncbi.nlm.nih.gov/pubmed/20831773",
                },
              },
              Object {
                "code": "ECO:0000313",
                "source": Object {
                  "alternativeUrl": "https://europepmc.org/abstract/MED/21500874",
                  "id": "21500874",
                  "name": "pubmed",
                  "url": "https://www.ncbi.nlm.nih.gov/pubmed/21500874",
                },
              },
              Object {
                "code": "ECO:0000313",
                "source": Object {
                  "alternativeUrl": "https://europepmc.org/abstract/MED/24829003",
                  "id": "24829003",
                  "name": "pubmed",
                  "url": "https://www.ncbi.nlm.nih.gov/pubmed/24829003",
                },
              },
              Object {
                "code": "ECO:0000313",
                "source": Object {
                  "alternativeUrl": "https://www.ensembl.org/homo_sapiens/Variation/Explore?v=RCV000813633",
                  "id": "RCV000813633",
                  "name": "ClinVar",
                  "url": "https://www.ncbi.nlm.nih.gov/clinvar/RCV000813633",
                },
              },
            ],
            "name": "Alzheimer disease (AD)",
          },
        ],
        "begin": "51",
        "clinicalSignificances": Array [
          Object {
            "sources": Array [
              "ClinVar",
            ],
            "type": "Variant of uncertain significance",
          },
        ],
        "codon": "AAG/AAT",
        "consequenceType": "missense",
        "cytogeneticBand": "21q21.3",
        "descriptions": Array [
          Object {
            "sources": Array [
              "ClinVar",
            ],
            "value": "Alzheimer disease",
          },
        ],
        "end": "51",
        "genomicLocation": "NC_000021.9:g.26112051C>A",
        "locations": Array [
          Object {
            "loc": "p.Lys51Asn",
            "seqId": "ENST00000346798",
            "source": "Ensembl",
          },
          Object {
            "loc": "c.153G>T",
            "seqId": "ENST00000346798",
            "source": "Ensembl",
          },
        ],
        "mutatedType": "N",
        "populationFrequencies": Array [
          Object {
            "frequency": 0.0002,
            "populationName": "MAF",
            "source": "1000Genomes",
          },
        ],
        "predictions": Array [
          Object {
            "predAlgorithmNameType": "PolyPhen",
            "predictionValType": "probably damaging",
            "predictorType": "multi coding",
            "score": 0.983,
            "sources": Array [
              "Ensembl",
            ],
          },
          Object {
            "predAlgorithmNameType": "PolyPhen",
            "predictionValType": "probably damaging",
            "predictorType": "multi coding",
            "score": 0.954,
            "sources": Array [
              "Ensembl",
            ],
          },
        ],
        "somaticStatus": 0,
        "sourceType": "large_scale_study",
        "type": "VARIANT",
        "wildType": "K",
        "xrefs": Array [
          Object {
            "id": "rs529782627",
            "name": "1000Genomes",
            "url": "https://www.ensembl.org/homo_sapiens/Variation/Explore?v=rs529782627",
          },
          Object {
            "id": "rs529782627",
            "name": "ExAC",
            "url": "https://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?type=rs&rs=rs529782627",
          },
          Object {
            "id": "rs529782627",
            "name": "gnomAD",
            "url": " http://gnomad.broadinstitute.org/awesome?query=rs529782627",
          },
        ],
      },
      "descriptions": Array [
        Object {
          "sources": Array [
            "ClinVar",
          ],
          "value": "Alzheimer disease",
        },
      ],
      "end": "51",
      "genomicLocation": "NC_000021.9:g.26112051C>A",
      "hasPredictions": true,
      "locations": Array [
        Object {
          "loc": "p.Lys51Asn",
          "seqId": "ENST00000346798",
          "source": "Ensembl",
        },
        Object {
          "loc": "c.153G>T",
          "seqId": "ENST00000346798",
          "source": "Ensembl",
        },
      ],
      "mutatedType": "N",
      "populationFrequencies": Array [
        Object {
          "frequency": 0.0002,
          "populationName": "MAF",
          "source": "1000Genomes",
        },
      ],
      "predictions": Array [
        Object {
          "predAlgorithmNameType": "PolyPhen",
          "predictionValType": "probably damaging",
          "predictorType": "multi coding",
          "score": 0.983,
          "sources": Array [
            "Ensembl",
          ],
        },
        Object {
          "predAlgorithmNameType": "PolyPhen",
          "predictionValType": "probably damaging",
          "predictorType": "multi coding",
          "score": 0.954,
          "sources": Array [
            "Ensembl",
          ],
        },
      ],
      "protvistaFeatureId": "uuid_id",
      "somaticStatus": 0,
      "sourceType": "large_scale_study",
      "start": "51",
      "type": "VARIANT",
      "variant": "N",
      "wildType": "K",
      "xrefNames": Array [
        "1000Genomes",
        "ExAC",
        "gnomAD",
      ],
      "xrefs": Array [
        Object {
          "id": "rs529782627",
          "name": "1000Genomes",
          "url": "https://www.ensembl.org/homo_sapiens/Variation/Explore?v=rs529782627",
        },
        Object {
          "id": "rs529782627",
          "name": "ExAC",
          "url": "https://www.ncbi.nlm.nih.gov/SNP/snp_ref.cgi?type=rs&rs=rs529782627",
        },
        Object {
          "id": "rs529782627",
          "name": "gnomAD",
          "url": " http://gnomad.broadinstitute.org/awesome?query=rs529782627",
        },
      ],
    },
    Object {
      "accession": "NC_045512.2:g.21568T>A",
      "alternativeSequence": "L",
      "begin": "2",
      "codon": "TTT/TTA",
      "consequenceType": "missense",
      "data": Object {
        "alternativeSequence": "L",
        "begin": "2",
        "codon": "TTT/TTA",
        "consequenceType": "missense",
        "end": "2",
        "genomicLocation": "NC_045512.2:g.21568T>A",
        "locations": Array [
          Object {
            "loc": "p.Phe2Leu",
            "seqId": "ENSSAST00005000004",
            "source": "EnsemblViruses",
          },
          Object {
            "loc": "c.6T>A",
            "seqId": "ENSSAST00005000004",
            "source": "EnsemblViruses",
          },
        ],
        "mutatedType": "L",
        "somaticStatus": 0,
        "sourceType": "large_scale_study",
        "type": "VARIANT",
        "wildType": "F",
        "xrefs": Array [
          Object {
            "id": "MN908947.3:21568:T:A",
            "name": "ENA",
          },
        ],
      },
      "end": "2",
      "genomicLocation": "NC_045512.2:g.21568T>A",
      "hasPredictions": undefined,
      "locations": Array [
        Object {
          "loc": "p.Phe2Leu",
          "seqId": "ENSSAST00005000004",
          "source": "EnsemblViruses",
        },
        Object {
          "loc": "c.6T>A",
          "seqId": "ENSSAST00005000004",
          "source": "EnsemblViruses",
        },
      ],
      "mutatedType": "L",
      "protvistaFeatureId": "uuid_id",
      "somaticStatus": 0,
      "sourceType": "large_scale_study",
      "start": "2",
      "type": "VARIANT",
      "variant": "L",
      "wildType": "F",
      "xrefNames": Array [
        "ENA",
      ],
      "xrefs": Array [
        Object {
          "id": "MN908947.3:21568:T:A",
          "name": "ENA",
        },
      ],
    },
  ],
}
`;
